orienteer-core/src/main/java/org/orienteer/core/component/table/filter/sql/OQueryBuilder.java

```diff
@@ -0,0 +1,118 @@
+package org.orienteer.core.component.table.filter.sql;
+
+import com.github.raymanrt.orientqb.query.Query;
+import com.google.common.collect.Maps;
+import com.orientechnologies.orient.core.db.ODatabase;
+import com.orientechnologies.orient.core.db.document.ODatabaseDocument;
+import com.orientechnologies.orient.core.metadata.schema.OProperty;
+import com.orientechnologies.orient.core.metadata.schema.OType;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.util.io.IClusterable;
+import ru.ydn.wicket.wicketorientdb.utils.DBClosure;
+
+import java.text.SimpleDateFormat;
+import java.util.Collection;
+import java.util.Date;
+import java.util.Map;
+
+import static com.github.raymanrt.orientqb.query.Projection.projection;
+
+/**
+ * @author Vitaliy Gonchar
+ */
+public class OQueryBuilder implements IClusterable {
+
+    private final String className;
+
+    public OQueryBuilder(String className) {
+        this.className = className;
+    }
+
+    public String build(Map<IModel<OProperty>, IModel<?>> filteredValues) {
+        String sql = "select from " + className;
+        if (needGenerateNewSql(filteredValues.values())) {
+            sql = generateSql(filteredValues);
+        }
+        return sql;
+    }
+
+    @SuppressWarnings("unchecked")
+    private String generateSql(Map<IModel<OProperty>, IModel<?>> filteredValues) {
+        Query query = new Query().from(className);
```

Comment:
So, it's working for "browse" page. What's about support of filtering on CalculableWidget and widget for "table" visualizer?